packages/web/src/components/inbox-settings-modal/InboxSettingsModalNew.tsx

```diff
@@ -0,0 +1,196 @@
+  allTitle: 'Allow Messages from Everyone',
+  followeeTitle: 'People You Follow',
+  tipperTitle: 'Tip Supporters',
+  tippedArtistsTitle: "Artists You've Tipped",
+  followersTitle: 'Your Followers',
+  verifiedTitle: 'Verified Users'
+}
+
+const options = [
+  {
+    title: messages.followeeTitle,
+    value: ChatPermission.FOLLOWEES
+  },
+  {
+    title: messages.tipperTitle,
+    value: ChatPermission.TIPPERS
+  },
+  {
+    title: messages.tippedArtistsTitle,
+    value: ChatPermission.TIPPEES
+  },
+  {
+    title: messages.followersTitle,
+    value: ChatPermission.FOLLOWERS
+  },
+  {
+    title: messages.verifiedTitle,
+    value: ChatPermission.VERIFIED
+  }
+]
+
+type InboxSettingsFormValues = {
+  allowAll: boolean
+  [ChatPermission.FOLLOWEES]: boolean
+  [ChatPermission.TIPPERS]: boolean
+  [ChatPermission.TIPPEES]: boolean
+  [ChatPermission.FOLLOWERS]: boolean
+  [ChatPermission.VERIFIED]: boolean
+}
+
+export const InboxSettingsModalNew = () => {
+  const [isVisible, setIsVisible] = useModalState('InboxSettings')
+  const handleClose = useCallback(() => setIsVisible(false), [setIsVisible])
+
+  const {
+    localPermission,
+    doFetchPermissions,
+    permissionStatus,
+    showSpinner
+    // setAndSavePermissions
+  } = useSetInboxPermissions({
+    audiusSdk
+  })
+
+  const handleSave = useCallback(
+    (values: InboxSettingsFormValues) => {
+      window.alert(JSON.stringify(values, null, 2))
+      if (values.allowAll) {
+        // setAndSavePermissions({ permit: ChatPermission.ALL })
+      }
+      // TODO: setAndSavePermissions doesn't accept multiple permissions yet
+      // setAndSavePermissions({})
+      handleClose()
+    },
+    [handleClose]
+  )
+
+  // Fetch the latest permissions for the current user when the modal is made visible
+  // Note that this will trigger the following effect as well, causing the permission state to update
+  useEffect(() => {
+    doFetchPermissions()
+  }, [doFetchPermissions])
+
+  const initialValues = {
+    allowAll: localPermission === ChatPermission.ALL,
+    [ChatPermission.FOLLOWEES]: localPermission === ChatPermission.FOLLOWEES,
+    [ChatPermission.TIPPERS]: localPermission === ChatPermission.TIPPERS,
+    [ChatPermission.TIPPEES]: localPermission === ChatPermission.TIPPEES,
+    [ChatPermission.FOLLOWERS]: localPermission === ChatPermission.FOLLOWERS,
+    [ChatPermission.VERIFIED]: localPermission === ChatPermission.VERIFIED
+  }
+
+  return (
+    <Formik<InboxSettingsFormValues>
+      initialValues={initialValues}
+      onSubmit={handleSave}
+    >
+      {({ submitForm }) => (
+        <Modal onClose={handleClose} isOpen={isVisible} size='medium'>
+          <ModalHeader onClose={handleClose}>
+            <ModalTitle title={messages.title} icon={<IconMessage />} />
+          </ModalHeader>
+          <ModalContent>
+            <InboxSettingsModalFields />
+          </ModalContent>
+          <ModalFooter>
+            <Flex gap='xl' flex={1}>
```

Comment:
what is this `flex={1}`?

Reply:
makes it grow to fit the container. Could also use 100% width I think in this case

Reply:
i thought `Flex` had flex:1 by default but maybe i'm wrong

Reply:
idk pretty sure it didn't work without it, but don't remember anymore tbh 😅